web/src/stream_ui_updates.js

```diff
@@ -228,22 +229,37 @@ export function update_add_subscriptions_elements(sub) {
     const $button_element = $add_subscribers_container
         .find('button[name="add_subscriber"]')
         .expectOne();
-    const allow_user_to_add_subs = sub.can_add_subscribers;
+
+    // We should disable the widgets when stream is private or
+    // realm level setting does not allow user to add subscribers.
+    const allow_user_to_add_subs =
+        sub.can_add_subscribers && settings_data.user_can_subscribe_other_users();
 
     if (allow_user_to_add_subs) {
         $input_element.prop("disabled", false);
         $button_element.prop("disabled", false);
         $button_element.css("pointer-events", "");
         $input_element.popover("destroy");
     } else {
-        $input_element.prop("disabled", true);
+        $input_element.prop("contenteditable", false);
```

Comment:
I think this can be a prep commit as this does not seem related to this. User should not have been able to have the cursor inside the input when button is disabled.